src/main/java/reposense/parser/CsvParser.java

```diff
@@ -207,4 +226,9 @@ private String getRowContentAsRawString(final CSVRecord record) {
      * {@code record} and add created objects into {@code results}.
      */
     protected abstract void processLine(List<T> results, final CSVRecord record) throws ParseException;
+
+    /**
+     * Gets the number of columns expected in the header.
+     */
+    protected abstract int getHeaderSize();
```

Comment:
Since `HEADER_SIZE` is a feature that all its child have, instead of an abstract method, is it better to make it a protected attribute and the child only need to overwrite this attribute and no need to overwite the method again and again?

Reply:
Thanks for the suggestion.

What about keeping the expected header size as a private variable in `CsvParser` ? 
This variable can be set in the constructor of `CsvParser`.

Reply:
I agree with your point. Go ahead.